olwebsite-app/src/utils/test-utils.js

```diff
@@ -1,26 +1,51 @@
 import React from "react";
 import { Router } from "react-router-dom";
 import { createMemoryHistory } from "history";
-import { render } from "@testing-library/react";
+import { render, getByLabelText, fireEvent } from "@testing-library/react";
 
 // Including this to get extended Jest matchers for expect
 import "@testing-library/jest-dom/extend-expect";
 
 const defaultRoute = "/";
 const defaultOptions = {};
 
-const renderWithRouter = (
+export function renderWithRouter(
   ui,
   {
     route = defaultRoute,
     history = createMemoryHistory({ initialEntries: [route] })
   } = defaultOptions
-) => {
+) {
   return {
     ...render(<Router history={history}>{ui}</Router>),
     history
   };
-};
+}
+
+/* Grabs the wanted JSX from the Component Passed in.
+ *
+ * @param container - the DOM of the rendered component
+ * @param className - className of the JSX to search for
+ * @returns {@link JSX}
+ */
+export function getComponentFromDOM(container, className) {
+  return container.querySelector(className);
```

Comment:
While this doesn't may not be 100% in line with the philosophy of react-testing-library because it exposes implementation details, i.e. the className, it does remove the need to add data-test-id attributes to the components.

[https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change](url)

Reply:
Yeah I agree, I don't like querying by className since this can lead to tests breaking easily. The need for it here is that if I use react-testing-library getBy* here is will keep throwing an error since it won't be able to find the HTML (which is the intended purpose).

Reply:
You could use queryBy* instead and expect either that it's not in the document, or that the query returns null. The queryBy api returns null if no matching element is found and will not throw an error.

Reply:
I might use this instead then, keep stuff tight